tests/Generic.Repository.Test/Cache/CacheRepositoryIOAccessTest.cs

```diff
@@ -3,53 +3,55 @@
 using System.Collections.Generic;
 using System.Threading.Tasks;
 
-namespace Generic.Repository.Test.Cache
+namespace Generic.Repository.UnitTest.Cache
 {
     public abstract class CacheRepositoryIOAccessTest<T>
         : CacheRepositoryExceptionTest<T>
         where T : class
     {
+        private const int MaxIterations = 10;
+
         [Test]
         public async Task CacheAccess_StressInput()
         {
             var valid = true;
-            for (int j = 0; j <= 10; j++)
+            for (var j = 0; j <= MaxIterations; j++)
             {
-                var listaTarefas = new List<Task>();
+                var scheduledListTask = new List<Task>();
                 try
                 {
-                    var tarefaGeral = Task.Run(async () =>
+                    var mainTask = Task.Run(async () =>
                     {
-                        var listaTarefaIO = new List<Task>();
+                        var listTaskIO = new List<Task>();
 
-                        for (int i = 0; i <= 10; i++)
+                        for (var i = 0; i <= MaxIterations; i++)
                         {
                             Cache.ClearCache();
 
-                            var tarefa = Task.Run(async () => await Cache.AddGet<T>(default));
-                            listaTarefaIO.Add(tarefa);
+                            var @task = Task.Run(async () => await Cache.AddGet<T>(default).ConfigureAwait(false));
+                            listTaskIO.Add(@task);
 
-                            tarefa = Task.Run(async () => await Cache.AddSet<T>(default));
-                            listaTarefaIO.Add(tarefa);
+                            @task = Task.Run(async () => await Cache.AddSet<T>(default).ConfigureAwait(false));
+                            listTaskIO.Add(@task);
 
-                            tarefa = Task.Run(async () => await Cache.AddProperty<T>(default));
-                            listaTarefaIO.Add(tarefa);
+                            @task = Task.Run(async () => await Cache.AddProperty<T>(default).ConfigureAwait(false));
+                            listTaskIO.Add(@task);
 
-                            tarefa = Task.Run(async () => await Cache.AddAttribute<T>(default));
-                            listaTarefaIO.Add(tarefa);
+                            @task = Task.Run(async () => await Cache.AddAttribute<T>(default).ConfigureAwait(false));
+                            listTaskIO.Add(@task);
                         }
 
-                        await Task.WhenAll(listaTarefaIO);
+                        await Task.WhenAll(listTaskIO);
                     });
 
-                    listaTarefas.Add(tarefaGeral);
+                    scheduledListTask.Add(mainTask);
+
+                    await Task.WhenAll(scheduledListTask);
                 }
-                catch (Exception)
+                catch (Exception e)
                 {
-                    valid = false;
+                    throw e;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using 'throw;' to preserve the stack trace.](https://app.codacy.com/manual/guilhermecaixeta/Generic.Service.DotNetCore2.2/pullRequest?prid=4961311)